src/writeData/components/PluginAddToExistingConfiguration/Configure.tsx

```diff
@@ -0,0 +1,143 @@
+
+// Selectors
+import {getAll} from 'src/resources/selectors'
+import {getDataLoaders} from 'src/dataLoaders/selectors'
+
+// Actions
+import {setBucketInfo} from 'src/dataLoaders/actions/steps'
+import {
+  setTelegrafConfigDescription,
+  setTelegrafConfigID,
+  setTelegrafConfigName,
+} from 'src/dataLoaders/actions/dataLoaders'
+
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = PluginConfigurationStepProps & ReduxProps
+
+const ConfigureComponent: FC<Props> = props => {
+  const {
+    buckets,
+    onSetBucketInfo,
+    onSetTelegrafConfigDescription,
+    onSetTelegrafConfigID,
+    onSetTelegrafConfigName,
+    setIsValidConfiguration,
+    setPluginConfig,
+    telegrafConfigID,
+    telegrafs,
+  } = props
+
+  const selectedTelegraf = telegrafs?.find(
+    telegraf => telegraf.id === telegrafConfigID
+  )
+
+  if (!selectedTelegraf) {
+    setIsValidConfiguration(false)
+  }
+
+  const selectedName = selectedTelegraf
+    ? selectedTelegraf.name
+    : 'Select a configuration'
+
+  const handleSelectTelegrafConfiguration = (telegraf: Telegraf) => {
+    const targetBucketName = Array.isArray(telegraf?.metadata?.buckets)
+      ? telegraf.metadata.buckets[0]
+      : ''
+    const selectedBucket = buckets.find(
+      bucket => bucket.name === targetBucketName
+    )
+    if (selectedBucket) {
+      const {orgID, name, id} = selectedBucket
+      onSetBucketInfo(orgID, name, id)
+    }
+    onSetTelegrafConfigDescription(telegraf.description)
+    onSetTelegrafConfigID(telegraf.id)
+    onSetTelegrafConfigName(telegraf.name)
+    setPluginConfig('')
+    setIsValidConfiguration(true)
+  }
+
+  return (
+    <>
+      <div>Telegraf Agent Configuration</div>
+      <Dropdown
+        button={(active, onClick) => (
+          <Dropdown.Button
+            active={active}
+            onClick={onClick}
+            status={ComponentStatus.Default}
+          >
+            {selectedName}
+          </Dropdown.Button>
+        )}
+        menu={onCollapse => (
+          <Dropdown.Menu onCollapse={onCollapse}>
+            {telegrafs.map(telegraf => (
+              <Dropdown.Item
+                key={telegraf.id}
+                onClick={handleSelectTelegrafConfiguration}
+                selected={
+                  selectedTelegraf && telegraf.id === selectedTelegraf.id
+                }
+                value={telegraf}
+                testID={`telegraf-configuration--${telegraf.name}`}
+              >
+                {telegraf.name}
+              </Dropdown.Item>
+            ))}
+          </Dropdown.Menu>
+        )}
+        testID="plugin-add-to-configuration--select-telegraf"
+      />
+    </>
+  )
+}
+
+const mstp = (state: AppState) => {
+  const buckets = getAll<Bucket>(state, ResourceType.Buckets)
+  const {telegrafConfigID} = getDataLoaders(state)
+  const telegrafs = getAll<Telegraf>(state, ResourceType.Telegrafs)?.sort(
```

Comment:
i think this is a really good opportunity to use a memoized selector using reselect. let's you and me talk tomorrow about creating a selector here and using reselect so that it's not re-sorting on every render call.

Reply:
This is true. For a large number of telegrafs... there is a LOT of sorting going on.

I think I can get started on writing a memoized `getTelegrafs` selector that is already sorted.

Reply:
I have a solution!

First, it should be noted that getting a resource, in this case Telegrafs and Buckets, from state is relatively quick. It's just accessing a property in a giant nested object. Accessing is very fast. Therefore, we can't really do anything to make getting a resource from state any more efficiency. It's fine to keep the accessing part.

The part that should be optimized is the sorting. We do not want to re-sort an array of resources every time we access it from state. (Resources are stored in state unsorted, in the order they were added.)

We can use a `useEffect` that only runs on mounting to do the sorting. This `useEffect` will have no dependencies and will never run again on re-renders. Thus, our sorting will never be done more than once per component mount.

This particular component does not allow for adding more Telegrafs or Buckets. We can safely assume that one sorting done initially will include all of the Telegrafs and Buckets that are available to the user.

Reply:
>Therefore, we can't really do anything to make getting a resource from state any more efficient.

Probably true, but we can do a lot to make it less efficient, and we need to be on the guard for that.